### ISSUE TITLE
Improve error message for attempting to duplicate existing grid field

### ISSUE THREAD

**Issue by mdpiper** [landlab/landlab#1500]
An error message is shown when a user attempts to add a new field to a grid that already has a field of the same name. The message is terse. We should make the message more descriptive to help users understand what went wrong.

**Comment by mdpiper** [comment]
Here's an example:

```python
from landlab import RasterModelGrid


rmg = RasterModelGrid((3, 4), 10)
rmg.add_zeros('topographic__elevation', at = 'node')
rmg.add_zeros('topographic__elevation', at = 'node')
```


    ---------------------------------------------------------------------------

    FieldError                                Traceback (most recent call last)

    Input In [4], in <cell line: 1>()
    ----> 1 rmg.add_zeros('topographic__elevation', at = 'node')


    File /opt/tljh/user/envs/ivy/lib/python3.9/site-packages/landlab/field/graph_field.py:1299, in GraphFields.add_zeros(self, *args, **kwds)
       1262 def add_zeros(self, *args, **kwds):
       1263     """add_zeros(name, at='node', units='-', clobber=False)
       1264 
       1265     Create and add an array of values, initialized to 0, to the field.
       (...)
       1297     LLCATS: FIELDCR
       1298     """
    -> 1299     data = self.add_empty(*args, **kwds)
       1300     data.fill(0)
       1301     return data


    File /opt/tljh/user/envs/ivy/lib/python3.9/site-packages/landlab/field/graph_field.py:1194, in GraphFields.add_empty(self, *args, **kwds)
       1192 copy = kwds.pop("copy", False)
       1193 clobber = kwds.pop("clobber", False)
    -> 1194 return self.add_field(
       1195     name,
       1196     self.empty(at=loc, **kwds),
       1197     at=loc,
       1198     units=units,
       1199     copy=copy,
       1200     clobber=clobber,
       1201 )


    File /opt/tljh/user/envs/ivy/lib/python3.9/site-packages/landlab/field/graph_field.py:1112, in GraphFields.add_field(self, *args, **kwds)
       1109 ds = getattr(self, "at_" + at)
       1111 if not clobber and name in ds:
    -> 1112     raise FieldError("{name}@{at}".format(name=name, at=at))
       1114 dims = (at,)
       1115 if value_array.ndim > 1:


    FieldError: topographic__elevation@node

**Comment by mdpiper** [comment]
@BCampforts raised this point when teaching Landlab at our recent GSA short course.

**Comment by mdpiper** [comment]
Fixed with #1503.
